File: frontend/src/types/flight-search-result.ts
export interface FlightSegment {
  origin: string;
  destination: string;
  start_time: string;
  end_time: string;
}

export interface BookingRequest {
  method: 'GET' | 'POST';
  url: string;
  post_data?: string | null;
}

export interface Flight {
  /** Represents a single flight route that can consist of multiple segments */
  airline: string;
  airline_logo_url?: string | null;
  price: number;
  segments: FlightSegment[];
  return_segments?: FlightSegment[] | null;
  duration_minutes: number;
  booking_url?: string | null;
  booking_request?: BookingRequest | null;
}

export interface FlightSearchResult {
  /** 
   * Contains all flight results for a specific (date, origin, destination) combination 
   */
  date: string;
  return_date?: string | null;
  origin: string;
  destination: string;
  flights: Flight[];
}

File: backend/data/FlightSearchResult.py
from datetime import date, datetime
from typing import List, Literal

from pydantic import BaseModel

class FlightSegment(BaseModel):
    origin: str
    destination: str
    start_time: datetime
    end_time: datetime


class BookingRequest(BaseModel):
    method: Literal["GET", "POST"]
    url: str
    post_data: str | None = None


class Flight(BaseModel):
    """Represents a single flight route that can consist of multiple segments"""
    airline: str
    airline_logo_url: str | None = None
    price: float
    segments: List[FlightSegment]
    duration_minutes: int
    booking_url: str | None = None
    booking_request: BookingRequest | None = None


class FlightSearchResult(BaseModel):
    """Contains all flight results for a specific (date, origin, destination) combination"""
    date: date
    origin: str
    destination: str
    flights: List[Flight]
